File: Solutions/1-two-sum/two-sum.scala
object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        // Create a hash map to store the indices of the elements
        val map = scala.collection.mutable.Map[Int, Int]()
        var result: Array[Int] = Array()

        // Iterate through the nums array
        for (i <- nums.indices) {
            val complement = target - nums(i)
            
            // If the complement is found in the map, store the indices in the result
            if (map.contains(complement)) {
                result = Array(map(complement), i)
            }
            
            // Otherwise, add the current number and its index to the map
            map(nums(i)) = i
        }
        
        // Return the result (it will be set to the correct pair when found)
        result
    }
}

File: Solutions/1-two-sum/two-sum.cpp
class Solution {
public:
vector<int> twoSum(vector<int>& nums, int target) {
    vector<int> result(2);  // Array of size 2 to store indices
    
    // Brute force checking to find the solution
    for (int i = 0; i < nums.size(); i++) {
        for (int j = i + 1; j < nums.size(); j++) {
            if (nums[i] + nums[j] == target) {
                result[0] = i;
                result[1] = j;
                return result;
            }
        }
    }

    return {};  // If no solution is found, return an empty vector
}
};